backend/layers/processing/reprocess_dataset_metadata.py

```diff
@@ -0,0 +1,308 @@
+            h5ad_job = Process(
+                target=DatasetMetadataReprocess.update_h5ad,
+                args=(
+                    self.artifact_bucket,
+                    self.datasets_bucket,
+                    artifact_uris[DatasetArtifactType.H5AD],
+                    current_dataset_version,
+                    new_artifact_key_prefix,
+                    metadata_update,
+                ),
+            )
+            artifact_jobs.append(h5ad_job)
+            h5ad_job.start()
+        else:
+            self.logger.error(f"Cannot find labeled H5AD artifact uri for {current_dataset_version_id}.")
+            raise ProcessingFailed from None
+
+        if DatasetArtifactType.RDS in artifact_uris:
+            self.logger.info("Main: Starting thread for rds update")
+            rds_job = Process(
+                target=DatasetMetadataReprocess.update_rds,
+                args=(
+                    self.artifact_bucket,
+                    self.datasets_bucket,
+                    artifact_uris[DatasetArtifactType.RDS],
+                    new_artifact_key_prefix,
+                    current_dataset_version_id,
+                    metadata_update,
+                ),
+            )
+            artifact_jobs.append(rds_job)
+            rds_job.start()
+        elif current_dataset_version.status.rds_status == DatasetConversionStatus.SKIPPED:
+            pass
+        else:
+            self.logger.error(
+                f"Cannot find RDS artifact uri for {current_dataset_version_id}, and Conversion Status is not SKIPPED."
+            )
+            raise ProcessingFailed from None
+
+        if DatasetArtifactType.CXG in artifact_uris:
+            self.logger.info("Main: Starting thread for cxg update")
+            cxg_job = Process(
+                target=DatasetMetadataReprocess.update_cxg,
+                args=(
+                    self.artifact_bucket,
+                    self.datasets_bucket,
+                    artifact_uris[DatasetArtifactType.CXG],
+                    f"s3://{self.cellxgene_bucket}/{new_artifact_key_prefix}.cxg",
+                    metadata_update,
+                ),
+            )
+            artifact_jobs.append(cxg_job)
+            cxg_job.start()
+        else:
+            self.logger.error(f"Cannot find cxg artifact uri for {current_dataset_version_id}.")
+            raise ProcessingFailed from None
+
+        # blocking call on async functions before checking for valid artifact statuses
+        [j.join() for j in artifact_jobs]
+
+    def update_dataset_citations_in_collection(
+        self,
+        collection_version_id: CollectionVersionId,
+    ):
+        collection_version = self.business_logic.get_collection_version(collection_version_id)
+        doi = next((link.uri for link in collection_version.metadata.links if link.type == "DOI"), None)
+        for dataset in collection_version.datasets:
```

Comment:
we'll likely need some blocking / batching / threadpooling for this for larger collections, looking into it